reconcile/utils/gql.py

```diff
@@ -198,13 +193,39 @@ def get_resource(path: str) -> dict[str, Any]:
     return get_api().get_resource(path)
 
 
+class PersistentRequestsHTTPTransport(RequestsHTTPTransport):
+    def connect(self):
+        if self.session is None:
+            # Copied over from RequestsHTTPTransport
+            self.session = requests.Session()
+            # we did not implement this in our copy!
+            assert self.retries == 0
+
+    def close(self) -> None:
+        pass
+
+
 def _init_gql_client(url: str, token: Optional[str]) -> Client:
+    global _local_client
+    if not _local_client:
+        # Requests.Session is not threadsafe, hence create one session per thread
+        _local_client = threading.local()
```

Comment:
Do we need to account for race condition here?

Reply:
Good point, let me check.

Reply:
Moved the initialization to global scope